fix: do not run containers as root by default in Helm chart

megakoresh opened the pull request:
### SUMMARY
This allows the user to control bootstrap behavior to avoid running containers with uid 0. Fixes #13869 

Some additions to documentation might be needed. Let me know if that's the case.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A - No change in behavior

### TEST PLAN
N/A - Helm chart does not currently have tests

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [X] Has associated issue: #13869 
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Commit:
Helm: no running as root by default

craig-rueda commented on helm/superset/values.yaml:
```diff
@@ -21,15 +21,19 @@
+# Install additional packages and do any other bootstrap configuration in this script
+# For production clusters it's recommended to build own image with this step done in CI
+bootstrapScript: |
+  #!/bin/bash
```
Did we want to comment all of these lines? Typo?

megakoresh replied:
Yes. I switched default uid to 1000 which will cause these lines to fail if those lines are uncommented. The purpose is to show users that this is a place to perform such tasks, but also that they must make their environment less secure by setting runAsUser value to 0.

craig-rueda replied:
Understood. Can you please put it back to 0 and enable that script? Maybe just add some verbage indicating that this isn't something one should do in production? There's several folks that rely on that thing currently (including me 😀).

megakoresh replied:
Ok, this is now done.

megakoresh committed:
Maintain for backwards compatibility

Reverted uid and script to work same way as they previously did to maintain backwards compatibility.
Added clarification in comments that this is not a recommended production configuration.
